src/Stratis.Bitcoin/Consensus/ConsensusManger.cs

```diff
@@ -0,0 +1,70 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.Logging;
+using NBitcoin;
+using Stratis.Bitcoin.Base;
+using Stratis.Bitcoin.BlockPulling;
+using Stratis.Bitcoin.Configuration.Settings;
+using Stratis.Bitcoin.Utilities;
+
+namespace Stratis.Bitcoin.Consensus
+{
+    public class ConsensusManager
+    {
+        private readonly Network network;
+        private readonly ILogger logger;
+        private readonly IChainedHeaderTree chainedHeaderTree;
+        private readonly IChainState chainState;
+        private readonly ConsensusSettings consensusSettings;
+        private readonly ConcurrentChain concurrentChain;
+        private readonly ILookaheadBlockPuller lookaheadBlockPuller;
+
+        /// <summary>The current tip of the chain that has been validated.</summary>
+        public ChainedHeader Tip { get; private set; }
+
+        private Dictionary<uint256, List<Action>> blocksRequested;
+
+        private AsyncQueue<(ChainedHeader headerFrom, ChainedHeader headerTo)> toDownloadQueue;
+
+        public ConsensusManager(Network network, ILoggerFactory loggerFactory, IChainedHeaderTree chainedHeaderTree, IChainState chainState, ConsensusSettings consensusSettings, ConcurrentChain concurrentChain, ILookaheadBlockPuller lookaheadBlockPuller)
+        {
+            this.network = network;
+            this.chainedHeaderTree = chainedHeaderTree;
```

Comment:
CHT is private instance of consensus manager. It is not a component running on itself it is not injected with dependency injection.

Reply:
Yes I am aware, however then we will find it problematic to test CM on its own, we could hack a bit and allow to set CHT as an internal property.

Also to keep the implementation internal I had to add an interface

Reply:
Are you sure we will have problems with testing? Because I don't think there will be any problems with CM to normally initialise its CHT instance and use it the full power even during tests, because CHT is quite independent of anything else.

Reply:
If it is a problem ... I cant say right now, testing CHT in all the tests of CM perhaps not a bad idea might be significantly harder.
Perhaps we try first as a hard instance and if we find mocking  is better we just move to an injected interface.

Reply:
Yes, let's do it like that. Let's assume no problems and only solve them when we are hit back.

Reply:
If we would like to mock CHT we can always just set private property in the test using the reflection